pycloudlib/azure/util.py

```diff
@@ -0,0 +1,102 @@
+# This file is part of pycloudlib. See LICENSE file for license information.
+"""Azure Util Functions."""
+import logging
+
+from azure.common.client_factory import (get_client_from_cli_profile,
+                                         get_client_from_json_dict)
+from knack.util import CLIError
+
+
+logger = logging.getLogger(__name__)
+
+
+def get_client(resource, config_dict):
+    """Get azure client based on the give resource.
+
+    This method will first verify if we can get the client
+    by using the information provided on the login account
+    of the user machine. If the user is not logged into Azure,
+    we will try to get the client from the ids given by the
+    user to this class.
+
+    Args:
+        resource: Azure Resource, An Azure resource that we want to get
+                    a client for.
+        config_dict: dict, Id parameters passed by the user to this class.
+
+    Returns:
+        The client for the resource passed as parameter.
+
+    """
+    try:
+        return get_client_from_cli_profile(resource)
+    except CLIError:
+        logger.debug(
+            "No valid azure-cli config found. Trying explicit config params"
+        )
+
+    required_keys = frozenset(
+        {"clientId", "clientSecret", "tenantId", "subscriptionId"}
+    )
+    missing_keys = required_keys.difference(set(config_dict.keys()))
+    if missing_keys:
+        raise RuntimeError(
+            "No AZ cli config found, missing required keys: {}".format(
+                ", ".join(missing_keys)
+            )
+        )
+
+    parameters = {
+        **config_dict,
```

Comment:
This merge will work unless someone provides and override of activeDirectoryEndpointUrl in config_dict right?

I think we still need parameters.update(config_dict) as config_dict should override local defaults if desired.

If the following were the case callers of  get_client would not be able to override any of the default params if desired.
>>> config_dict = {'a': 'override_a'}
>>> parameters = {**config_dict, 'a': 'default_a'}
>>> parameters
{'a': 'default_a'}

Reply:
Although we are not currently supporting overriding these fields, you are right that this will pose a problem if we want to support that. I will update the code